File: ruby solutions/pascal_triangle.rb
# @param {Integer} num_rows
# @return {Integer[][]}
def generate(num_rows)
    rows = []
    (0...num_rows).each { |i|
        rows[i] = []
        (0..i).each { |j|
        rows[i][j] = (j == 0 || j == i ? 1 : rows[i-1][j-1] + rows[i-1][j])
        }
    }
    rows
end

File: python solutions/pascal_triangle.py
from typing import List


class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        rows = []
        if numRows > 0: rows.append([1])
        if numRows > 1: rows.append([1, 1])
        for i in range(3, numRows+1):
            temp = [1]
            for j in range(i-2):
                temp.append(rows[i-2][j] + rows[i-2][j+1])
            temp.append(1)
            rows.append(temp)
        return rows